### ISSUE TITLE
AspectRegistryEvent is unusable in the obfuscated game environment

### ISSUE THREAD

**Issue by cpw** [Azanor/thaumcraft-api#38]
This class is not able to be successfully loaded and processed into a Forge Event when the game is running in an obfuscated environment (such as the regular game client).

This is probably due to the fact that you're forcing the recomputation of Java StackFrames, which forces ASM to do class loading for `net.minecraft.item.ItemStack` which doesn't exist in the context that ASM sees (it's generated by the Deobfuscation code).

In summary, you should push the actual implementing code outside the Event's immediate class scope and just call it. (Make the event return a proxy object for consumers, put all the code in the proxy object?)

It's amazing what we can still uncover..

**Comment by Drullkus** [comment]
Here's a test build of Twilight Forest in case you need a mod instance using your event: 
http://builds.steamnsteel.info/browse/TF-ONETWELVE-606/artifact/JOB1/Universal/twilightforest-1.12.2-3.8.606-universal.jar

**Comment by Azanor** [comment]
Doh. Thanks
